Include/pystate.h

```diff
@@ -101,9 +115,7 @@ typedef struct _ts {
     PyObject *curexc_value;
     PyObject *curexc_traceback;
 
-    PyObject *exc_type;
-    PyObject *exc_value;
-    PyObject *exc_traceback;
+    PyExcState exc_state;
```

Comment:
Given that this is already quite an invasive change, maybe replace `curexc_*` fields with `curexc_state`?